### PR TITLE
New sample: Display OGC API collection

### PR DESCRIPTION
**Pull request by jenmerritt**
Adds sample as per the design.

### CODE REVIEW

File: lib/samples/display_ogc_api_collection/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+1. Create an `OgcFeatureCollectionTable` object using a URL to an OGC API feature service and a collection ID.
+2. Set the feature table's `FeatureRequestMode` property to `FeatureRequestMode.manualCache`.
+3. Call `OgcFeatureCollectionTable.load()`.
+4. Create a `FeatureLayer` using the feature table and add it to the map view.
+5. Every time the map view navigation completes:
+   1. Create `QueryParameters`.
+   2. Set the parameter's `Geometry` to the current extent of the map view.
+   3. Set the parameter's `SpatialRelationship` property to `SpatialRelationship.intersects`.
+   4. Set the `maxFeatures` property to 5000 (some services have a low default value for maximum features).
+   5. Call `OgcFeatureCollectionTable.PopulateFromService()` using the query parameters from the previous steps.
```
Comment (PaulAllanSturm) [inline lib/samples/display_ogc_api_collection/README.md:26]: "PopulateFromService()" should be "populateFromService()" -- initial lowercase letter.

I think the various references to properties should be initial-lowercase as well -- featureRequestMode, geometry, spatialRelationship. It's a bit picky, but when referring to the property it should be lowercase, and when referring to the type it should be uppercase.

File: lib/samples/display_ogc_api_collection/display_ogc_api_collection.dart [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    );
+  }
+
+  void onMapViewReady() async {
+    // Create a map with a basemap style and add to the map view controller.
+    final map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISTopographic);
+    _mapViewController.arcGISMap = map;
+
+    // Create an OGC feature collection table by passing in a service URL and a collection id.
+    final serviceUri = Uri.parse('https://demo.ldproxy.net/daraa');
```
Comment (PaulAllanSturm) [inline lib/samples/display_ogc_api_collection/display_ogc_api_collection.dart:75]: As an aside, I've been thinking we should declare URL strings as `const`, and then move the call to `Uri.parse()` down into where it gets used. I think that ends up being cleaner overall, especially with longer URLs.

File: lib/samples/display_ogc_api_collection/display_ogc_api_collection.dart [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        }
+        // Set the ready state variable to true to enable the sample UI.
+        setState(() => _ready = true);
+      }
+    });
+
+    // Zoom to a small area within the dataset by default.
+    final datasetExtent = ogcFeatureCollectionTable.extent;
+    if (datasetExtent != null) {
+      _mapViewController.setViewpointAnimated(
```
Comment (PaulAllanSturm) [inline lib/samples/display_ogc_api_collection/display_ogc_api_collection.dart:132]: Why setViewpointAnimated? I think elsewhere we'd just set the initialViewpoint.

File: lib/samples/display_ogc_api_collection/display_ogc_api_collection.dart [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        if (_mapViewController.visibleArea != null) {
+          // Create a query based on the current visible extent.
+          // Set a limit of 5000 on the number of returned features per request,
+          // because the default on some services could be as low as 10.
+          final queryParameters = QueryParameters()
+            ..geometry = _mapViewController.visibleArea!.extent
+            ..spatialRelationship = SpatialRelationship.intersects
+            ..maxFeatures = 5000;
+          // Populate the table with the query, leaving existing table entries intact.
+          // Setting outFields to null requests all fields.
```
Comment (PaulAllanSturm) [inline lib/samples/display_ogc_api_collection/display_ogc_api_collection.dart:117]: Should "Setting outFields to null" be "Setting outFields to an empty array"?

File: lib/samples/display_ogc_api_collection/display_ogc_api_collection.dart (hunk 3 above)
Comment (jenmerritt) [reply lib/samples/display_ogc_api_collection/display_ogc_api_collection.dart:132]: It needs to trigger the navigation changed event to populate the OGC table. It's part of the sample design.

File: lib/samples/display_ogc_api_collection/display_ogc_api_collection.dart (hunk 4 above)
Comment (jenmerritt) [reply lib/samples/display_ogc_api_collection/display_ogc_api_collection.dart:117]: Ah yep, that's a copy paste miss, thanks.

**Review by PaulAllanSturm — APPROVED** [review]

**Pull request merged**